app/tests/unit/forms/submission/controller.spec.js

```diff
@@ -99,26 +111,48 @@ describe('templateUploadAndRender', () => {
       id: '123',
     },
     submission: {
+      confirmationId: confirmationId,
+      id: submissionId,
       submission: {
-        submission: {
-          data: {
-            simpletextfield: 'fistName lastName',
-            submit: true,
-          },
+        data: {
+          simpletextfield: 'firstName lastName',
+          submit: true,
```

Comment:
This was previously nested too deeply and not appearing in what was sent to the CDOGS service! The tests now check that the data returned from the "database" is properly sent to "cdogs".